src/box/memtx_engine.c

```diff
@@ -170,12 +178,13 @@ memtx_engine_recover_snapshot(struct memtx_engine *memtx,
 	int rc;
 	struct xrow_header row;
 	uint64_t row_count = 0;
+	int is_not_system_space = -1;
```

Comment:
Why don't use boolean type?

Reply:
Also would be nice to see explanation why we have to stop recovery if one of system space's rows is broken.

Reply:
We can't use boolean type here because there is a case where xlog_cursor_next return not 0 imidiatly. This means that snapshoot system data is strpongly corrupted. in this case we check that is_not_system_space  < 0 later (it's not changed and == -1) and  return from this function with fail.